### DISCUSSION TITLE
How to create a dependency graph?

### DISCUSSION THREAD

**Discussion by thiromi** [meadsteve/lagom#247] [Q&A]
I wanted to build the dependencies graph using maybe graphviz or another library that would create graphs. Before I start digging deeper in the source code to check this, I wanted to know if someone has already done something similar

**Comment by meadsteve** [comment]
I'm not aware of anyone having done this. I certainly haven't. It sounds really interesting though. I'd love to see how you get on.

**Reply by thiromi** [reply]
I created a gist (without using lagom) that inspects one class and create a dependency tree from its dependencies: https://gist.github.com/thiromi/75b23ccaf02844c5a97ce1507189bd40

still not optimal because it has some flaws:

* doesn't show the concrete classes we've set up in lagom
* doesn't stop on the `Container` type (for awaitable dependencies, I've added the `Container` class as a dependency to be resolved afterward, and this class has `typing.Optional[ForwardRef('Container')]` and `typing.Union[bool, logging.Logger]` as dependencies)
* dependencies resolved in the class (like the awaited ones or when someone uses lagom as a service locator) won't be listed there
